rfc/006-semantic-versioning.md

```diff
@@ -0,0 +1,76 @@
+# RFC 006: Semantic Versioning
+
+## Changelog
+
+- 2021-11-19: Initial Draft
+
+## Author(s)
+
+- Callum Waters @cmwaters
+
+## Context
+
+We use versioning as an instrument to hold a set of promises to users and signal when such a set changes and how. In the conventional sense of a Go library, major versions signal that the public Go API’s have changed in a breaking way and thus require the users of such libraries to change their usage accordingly. Tendermint is a bit different in that there are multiple users: application developers (both in-process and out-of-process), node operators, and external clients. More importantly, both how these users interact with Tendermint and what's important to these users differs from how users interact and what they find important in a more conventional library.
```

Comment:
To clarify, is this document solely about the Go project? If so, should it be in the Go repo? or is it about spec-specific changes? If we're talking about cross-chain data/ relayers etc, I would say this belongs here in the spec repo, but I wanted to make sure I was understanding correctly.

Reply:
I wanted the semantic versioning to be something that all Tendermint implementations adopt hence why I decided to have this in the spec repo (although I also thought about writing this in the tendermint go repo).

Reply:
I think that this makes sense. We keep definitions for the public APIs in this repo so it makes sense to have a document describing how we version them live alongside them. It means tho, that we should probably come up with a more disciplined process for 'releasing' this repo.

Reply:
Yeah I totally agree